typescript/sdk/src/core/TestRecipientDeployer.ts

```diff
@@ -39,7 +38,28 @@ export class TestRecipientDeployer extends HyperlaneDeployer<
     chain: ChainName,
     config: TestRecipientConfig,
   ): Promise<TestRecipientContracts> {
-    const testRecipient = await this.deployContract(chain, 'testRecipient', []);
+    const predeployed = this.readCache(
+      chain,
+      this.factories['testRecipient'],
+      'testRecipient',
+    );
+    let usePreviousDeployment = false;
+    if (
+      predeployed &&
+      eqAddress(
+        await predeployed.owner(),
+        await this.multiProvider.getSignerAddress(chain),
+      )
+    ) {
```

Comment:
Per our chat I lean _slightly_ towards removing this check but I'm okay either way :)